.github/workflows/ci-pipeline.yaml

```diff
@@ -0,0 +1,38 @@
+name: Continuous Integration
+on: 
+  push:
+    paths-ignore:
+      -'./README.md'
+      -'./tricks.yaml'
+      -'documentation/**'
+      -'logs/**'
+  
+  pull_request:
+    paths-ignore:
+      -'./README.md'
+      -'./tricks.yaml'
+      -'documentation/**'
+      -'logs/**'
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      # only output hash of image and pass that to docker run
+      - run: docker run --rm $(docker build -q . --target testing)
```

Comment:
While this is nice and concise, I'd be inclined to split the build & run into two steps to make it easier to digest what's happening; this format is nicer when things go well, but potentially a bit more obscure when things are going wrong!